### ISSUE TITLE
How to get the menu's direction?

### ISSUE THREAD

**Issue by taylor-work** [szhsin/react-menu#837]
React/React-dom version: 17.0.2
React-Menu version: 3.0.0

I have a scenario where I want the submenu's align to change based on the parent's menu direction. Is there a way to get the parent's menu direction? 

I know I can pass in the direction to the parent menu however when the parent menu changes direction based on the bounding box I no longer control the parent's menu direction.

**Comment by szhsin** [comment]
@taylor-work , it is not possible to get parent menu's direction currently, unless in the functional form of `menuClassName` prop.

This can be enhanced by allowing menu's `children` to accept render prop. Therefore, would the following API satisfy your use case?
```jsx
<Menu menuButton={<MenuButton>Menu</MenuButton>}>
  {({ dir }) => (
    <SubMenu label="Edit" align={dir === 'top' ? 'end' : 'start'}>
      <MenuItem>Cut</MenuItem>
      <MenuItem>Copy</MenuItem>
      <MenuItem>Paste</MenuItem>
    </SubMenu>
  )}
</Menu>
```
The `dir` will be the final computed direction after bounding box check.

**Comment by taylor-work** [comment]
That would work for our use case. Thank you!

**Comment by szhsin** [comment]
Released in [v3.5.0](https://github.com/szhsin/react-menu/releases/tag/v3.5.0)
